Breeze.TumbleBit.Client/TumbleBitManager.cs

```diff
@@ -561,6 +542,13 @@ public enum TumbleState
 
             this.runtime = await TumblerClientRuntime.FromConfigurationAsync(config).ConfigureAwait(false);
 
+            // Check if origin wallet has a sufficient balance to begin tumbling at least 1 cycle
+            if (!runtime.HasEnoughFundsForCycle())
+            {
+                this.logger.LogDebug("Insufficient funds in origin wallet");
+                throw new Exception("Insufficient funds in origin wallet");

```

Comment:
Throwing generic `Exception` is always a bad practice. Can this be changed to something a bit more specific. Alternatively, there is always `ApplicationException`, which is still better than `Exception`. Ideally we should have dedicated exception for this.